ui/base/ime/input_method_minimal_unittest.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ui/base/ime/input_method_minimal.h"
+
+#include "base/macros.h"
+#include "base/memory/singleton.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "ui/base/ime/dummy_text_input_client.h"
+#include "ui/base/ime/input_method_delegate.h"
+#include "ui/base/ime/input_method_initializer.h"
+#include "ui/events/event.h"
+
+namespace ui {
+namespace {
+
+class InputMethodMinimalTest : public internal::InputMethodDelegate,
+                               public testing::Test,
+                               public DummyTextInputClient {
+ protected:
+  InputMethodMinimalTest()
+      : input_method_minimal_(NULL),
+        last_key_code_(VKEY_UNKNOWN),
+        stop_propagation_post_ime_(false) {}
+  ~InputMethodMinimalTest() override {}
+
+  void SetUp() override {
+    input_method_minimal_ = new InputMethodMinimal(this);
+    input_method_minimal_->OnFocus();
+  }
+
+  void TearDown() override {
+    delete input_method_minimal_;
+    input_method_minimal_ = NULL;
+  }
+
+  // Overridden from ui::internal::InputMethodDelegate:
+  ui::EventDispatchDetails DispatchKeyEventPostIME(
+      ui::KeyEvent* key_event) override {
+    last_key_code_ = key_event->key_code();
+    if (stop_propagation_post_ime_)
+      key_event->StopPropagation();
+    return ui::EventDispatchDetails();
+  }
+
+  // Overridden from ui::TextInputClient:
+  void InsertChar(const ui::KeyEvent& event) override {
+    DummyTextInputClient::InsertChar(event);
```

Comment:
You don't need to inherit DummyTextInputClient in this case. Just declare it bellow as 
DummyTextInputClient text_input_client_; 

then below in StopPropagationTest, just provide it as text_input_client_.get().

Reply:
You're right. I declared inside StopPropagationTest because I think InputMethodMinimalTest doesn't need to include it and it's dependent on each test case.